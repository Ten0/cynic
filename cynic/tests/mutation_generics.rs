#[allow(non_snake_case, non_camel_case_types)]
mod schema {
    cynic::use_schema!("../schemas/raindancer.graphql");
}

#[derive(cynic::QueryVariables, Debug)]
pub struct SignInVariables<'a> {
    pub input: SignInInput<'a>,
}

#[derive(cynic::InputObject, Debug)]
#[cynic(schema_path = "../schemas/raindancer.graphql")]
pub struct SignInInput<'a> {
    pub password: &'a str,
    pub username: String,
}

#[derive(cynic::QueryFragment, Debug)]
#[cynic(
    graphql_type = "MutationRoot",
    variables = "SignInVariables",
    schema_path = "../schemas/raindancer.graphql"
)]
pub struct SignIn {
    #[arguments(input: $input)]
    pub sign_in: String,
}

#[test]
fn test_query_building() {
    use cynic::MutationBuilder;

    let operation = SignIn::build(SignInVariables {
        input: SignInInput {
            password: "password?",
            username: "username".to_owned(),
        },
    });

    insta::assert_snapshot!(operation.query);
}
