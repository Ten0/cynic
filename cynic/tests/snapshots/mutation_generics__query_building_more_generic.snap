---
source: cynic/tests/mutation_generics.rs
assertion_line: 75
expression: operation.query
---
mutation($input: SignInInput!) {
  signIn(input: $input)
}


